pkg/netpol/eval/resources.go

```diff
@@ -78,7 +78,26 @@ func NewPolicyEngineWithObjects(objects []scan.K8sObject) (*PolicyEngine, error)
 			return nil, err
 		}
 	}
-	return pe, nil
+	err = pe.resolveMissingNamespaces()
+	return pe, err
+}
+
+func (pe *PolicyEngine) resolveMissingNamespaces() error {
+	for _, pod := range pe.podsMap {
+		ns := pod.Namespace
+		if _, ok := pe.namspacesMap[ns]; !ok {
+			// create a ns object and upsert to PolicyEngine
+			nsObj := &corev1.Namespace{}
```

Comment:
Using an object initialization syntax here might be more succinct:
```go
nsObj := &corev1.Namespace {
   Name: ns,
   Labels: map[string]string{
         "kubernetes.io/metadata.name": ns,
   },
}
```

Reply:
changed, thanks.